sdk/src/wallet/bindings/java/lib/native/src/lib.rs

```diff
@@ -194,25 +194,24 @@ pub unsafe extern "system" fn Java_org_iota_api_NativeApi_sendMessage(
 pub unsafe extern "system" fn Java_org_iota_api_NativeApi_listen(
     mut env: JNIEnv,
     _class: jclass,
-    events: JObjectArray,
+    events: JIntArray,
     callback: jobject,
 ) -> jstring {
     env_assert!(env, std::ptr::null_mut());
 
-    let string_count = jni_err_assert!(env, env.get_array_length(&events), std::ptr::null_mut());
+    let count = jni_err_assert!(env, env.get_array_length(&events), std::ptr::null_mut());
 
-    // Safe cast as we dont have that many events
-    let mut events_list: Vec<WalletEventType> = Vec::with_capacity(string_count as usize);
+    // Safe cast as we don't have that many events
+    let mut events_list: Vec<WalletEventType> = Vec::with_capacity(count as usize);
+    let mut events_integers = vec![0; count as usize];
```

Comment:
Can't use an array here?

Reply:
Sir this is Java, I just made it compile

Reply:
hes improving java, what a time to be alive!